common-lib/src/main/scala/com/gu/mediaservice/lib/aws/SqsHelpers.scala

```diff
@@ -0,0 +1,31 @@
+package com.gu.mediaservice.lib.aws
+
+import com.amazonaws.services.sqs.model.{Message => SQSMessage}
+import com.gu.mediaservice.lib.net.URI.{decode => uriDecode}
+import play.api.libs.json.Json
+
+import scala.util.Try
+
+trait SqsHelpers {
+  def getApproximateReceiveCount(message: SQSMessage): Int =
+   Try(message.getAttributes.get("ApproximateReceiveCount").toInt).toOption.getOrElse(-1)
+
+  def extractS3KeyFromSqsMessage(message: SQSMessage): Try[String] = Try {
+
+    // we only need the key, the full message shape can be seen at
+    // https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html
+
+    val s3Records = Json.parse(message.getBody) \ "Records" \\ "s3"
+
+    if(s3Records.size != 1) {
```

Comment:
This works, but I think you could avoid the slightly unpleasant `.head` with something like:
```
s3Records match {
  case h :: Nil => uriDecode((h \ "object" \ "key").as[String])
  case _ => throw new Exception(s"Expected 1 record containing 's3' in message body, got ${s3Records.size}")
}
```

Reply:
this is much more elegant - thanks - we've refactored now